docs/conf.py

```diff
@@ -361,3 +373,260 @@ def _skip(self, word):
+                    continue
+
+                ref_str = ref[20:].decode()
+                if self._release_branch_pattern.match(ref_str):
+                    v = Version(ref_str)
+                    return Version("{}.{}".format(v.major, v.minor + 1))
+                elif self._dev_branch_pattern.match(ref_str):
+                    major, _, _ = ref_str.partition(".")
+                    return Version("{}.0.0".format(int(major) + 1))
+        return None
+
+    def run(self):
+        # type: () -> nodes.Node
+        """
+        Run to generate the output from .. ddtrace-release-notes:: directive
+
+
+        1. Determine the max version cutoff we need to report for
+
+           We determine this by traversing the git log until we
+           find the first dev or release branch ref.
+
+           If we are generating for 1.x branch we will use 2.0 as the cutoff.
+
+           If we are generating for 0.60 branch we will use 0.61 as the cutoff.
+
+           We do this to ensure if we are generating notes for older versions
+           we do no include all up to date release notes. Think releasing 0.57.2
+           when there is 0.58.0, 0.59.0, 1.0.0, etc we only want notes for < 0.58.
+
+        2. Iterate through all release branches
+
+           A release branch is one that matches the ``^[0-9]+.[0-9]+``` pattern
+
+           Skip any that do not meet the max version cutoff.
+
+        3. Determine the earliest version to report for each release branch
+
+           If the release has only RC releases then use ``.0rc1`` as the earliest
+           version. If there are non-RC releases then use ``.0`` version as the
+           earliest.
+
+           We do this because we want reno to only report notes that are for that
+           given release branch but it will collapse RC releases if there is a
+           non-RC tag on that branch. So there isn't a consistent "earliest version"
+           we can use for in-progress/dev branches as well as released branches.
+
+
+        4. Generate a reno config for reporting and generate the notes for each branch
+        """
+        # This is where we will aggregate the generated notes
+        title = " ".join(self.content)
+        result = statemachine.ViewList()
+
+        # Determine the max version we want to report for
+        max_version = self._get_report_max_version()
+        LOG.info("capping max report version to %r", max_version)
+
+        # For each release branch, starting with the newest
+        for version, ref in self._release_branches:
+            # If this version is equal to or greater than the max version we want to report for
+            if max_version is not None and version >= max_version:
+                LOG.info("skipping %s >= %s", version, max_version)
+                continue
+
+            # Older versions did not have reno release notes
+            if (version.major, version.minor) < (0, 44):
```

Comment:
Do we actually have to code this since there shouldn't be any reno notes in those versions?

Reply:
I'll update the comment, but yes we do since we tell reno to go look for notes on each branch, and since there are none on this branch (and no "earlier version" available) it'll fail.